Lombiq.Vsix.Orchard/Services/LogWatcher/LogFileWatcherBase.cs

```diff
@@ -14,82 +14,91 @@ public abstract class LogFileWatcherBase : ILogFileWatcher
     {
         private const int DefaultLogWatcherTimerIntervalInMilliseconds = 1000;
 
-
         private readonly AsyncPackage _package;
         protected readonly ILogWatcherSettingsAccessor _logWatcherSettingsAccessor;
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "Usage",
+            "CA2213:Disposable fields should be disposed",
+            Justification = "The timer is Disposed from the StopWatching method when the LogFileWatcherBase class is Disposed")]
         private Timer _timer;
         private bool _isWatching;
         private ILogFileStatus _previousLogFileStatus;
 
-
         public event EventHandler<LogChangedEventArgs> LogUpdated;
 
-
         protected LogFileWatcherBase(AsyncPackage package, ILogWatcherSettingsAccessor logWatcherSettingsAccessor)
         {
             _package = package;
             _logWatcherSettingsAccessor = logWatcherSettingsAccessor;
         }
 
+        protected abstract Task<string> GetLogFileNameAsync();
 
-        protected abstract Task<string> GetLogFileName();
-
-        public virtual async System.Threading.Tasks.Task StartWatching()
+        public virtual async System.Threading.Tasks.Task StartWatchingAsync()
         {
             if (_isWatching) return;
 
-            _previousLogFileStatus = await GetLogFileStatus();
+            _previousLogFileStatus = await GetLogFileStatusAsync();
 
             // Using this pattern: https://stackoverflow.com/a/684208/220230 to prevent overlapping timer calls.
-            // Since Timer callbacks are executed in a ThreadPool thread 
+            // Since Timer callbacks are executed in a ThreadPool thread
             // (https://docs.microsoft.com/en-us/dotnet/standard/threading/timers) they won't block the UI thread.
-            _timer = new Timer(async state =>
+            _timer = new Timer(TimerCallback, null, 0, Timeout.Infinite);
+
+            _isWatching = true;
+        }
+
+        private void TimerCallback(object state) => _ = TimerCallbackAsync();

```

Comment:
Az összes ilyen discard-assignos `Task` ugyanúgy fire and forget, mint az `asnyc void`. Egyiket se kellene.

Reply:
Azért csináltam így, mert így ha egy exception lesz a Task-on belül, akkor az nem állítja le az egész process-t, hanem az Exception újra lesz dobva itt.

Reply:
Nincs awaitelve, és sync-ké téve. Így nem értem, miért ne lenne fire and forget, akkor is, ha valamiért ez a discardos megoldás valamennyire másképp működik.

Reply:
Food for thought: https://stackoverflow.com/questions/42838362/ignoring-the-return-value-from-an-async-task-method